chia/wallet/conditions.py

```diff
@@ -404,6 +404,48 @@ def from_program(cls, program: Program, puzzle_hash: Optional[bytes32] = None) -
         )
 
 
+@final
+@streamable
+@dataclass(frozen=True)
+class SendMessage(Condition):
+    mask: uint8
```

Comment:
The purpose of these condition classes is to be as ergonomic as possible for developers/users (who can pass these in via RPC).  See the announcement classes for an example where more than just the bare minimum for the condition is accepted and is transformed as necessary when asking for the condition as a program.

Something like:
```
class SendMessage(Condition):
  msg: bytes
  mask: Optional[uint8] = None
  requires_parent_id: bool = False
  requires_puzzle_hash: bool = False
  requires_amount: bool = False
  parent_id: Optional[bytes32] = None
  puzzle_hash: Optional[bytes32] = None
  amount: Optional[uint64] = None
```
with some relevant __post_init__ checks and `@property`s would be more in the spirit of the file.

Reply:
Same goes for ReceiveMessage as well obviously

Reply:
in this example, the `mask` would be computed, and not a field, right?

Reply:
also, I think it would make sense to make this more ergonomic when we need it. The node needs this feature months befor the wallet does. The reason I added these was because of the completeness test that started failing.

Reply:
Yeah if you want to punt on it that's fine but it does go against the _spirit_ of the completeness check.  (To answer the question about mask, no, you want to be able to construct the class with any of the information you have.  You might be parsing clvm and only pull out the mask and want to use that to construct the class. There would be an `@property` or function or something though to _get_ the mask if the other pieces of information have been supplied.)